core/src/test/java/org/elasticsearch/gateway/MetaDataWriteDataNodesIT.java

```diff
@@ -170,7 +171,9 @@ protected void assertIndexInMetaState(final String nodeName, final String indexN
 
     private boolean indexDirectoryExists(String nodeName, String indexName) {
         NodeEnvironment nodeEnv = ((InternalTestCluster) cluster()).getInstance(NodeEnvironment.class, nodeName);
-        for (Path path : nodeEnv.indexPaths(indexName)) {
+        ClusterService clusterService = ((InternalTestCluster) cluster()).getInstance(ClusterService.class, nodeName);
```

Comment:
same comment about resolving. maybe easier to change the method to get an Index object as a param?